source/nodejs/adaptivecards/src/carousel.ts

```diff
@@ -0,0 +1,264 @@
+import "swiper/css";
+import "swiper/css/pagination";
+import "swiper/css/navigation";
+import { GlobalSettings } from "./shared";
+
+// TODO - investigate/wrap
+Swiper.use([
+    Navigation,
+    Pagination,
+    Scrollbar,
+    A11y,
+    History,
+    Keyboard,
+    Autoplay
+]);
+
+export class CarouselPage extends Container {
+    //#region Schema
+
+    //#endregion
+
+    protected internalRender(): HTMLElement | undefined {
+
+        let swiperSlide: HTMLElement = document.createElement("div");
+        swiperSlide.className = this.hostConfig.makeCssClassName("swiper-slide");
+
+        let renderedElement = super.internalRender();
+        Utils.appendChild(swiperSlide, renderedElement);
+        return swiperSlide;
+    }
+    static unsupportedElementsList: Set<string>;
+
+    protected internalParse(source: any, context: SerializationContext) {
+        super.internalParse(source, context);
+
+        this.setShouldFallback(false);
+    }
+
+    protected shouldSerialize(context: SerializationContext): boolean {
+        return true;
+    }
+
+    getJsonTypeName(): string {
+        return "CarouselPage";
+    }
+
+    get isStandalone(): boolean {
+        return false;
+    }
+}
+
+export class Carousel extends ContainerBase {
+    //#region Schema
+
+    static readonly timerProperty = new NumProperty(Versions.v1_6, "timer", 0);
+    @property(Carousel.timerProperty)
+    timer?: number;
+
+    //#endregion
+
+    private _pages: CarouselPage[] = [];
+    private _renderedPages: CarouselPage[];
+
+    getJsonTypeName(): string {
+        return "Carousel";
+    }
+
+    getItemCount(): number {
+        return this._pages.length;
+    }
+
+    getPageAt(index: number): CarouselPage {
+        return this._pages[index];
+    }
+
+    getItemAt(index: number): CardElement {
+        return this.getPageAt(index);
+    }
+
+    removeItem(item: CardElement): boolean {
+        if (item instanceof CarouselPage) {
+            const itemIndex = this._pages.indexOf(item);
+
+            if (itemIndex >= 0) {
+                this._pages.splice(itemIndex, 1);
+
+                item.setParent(undefined);
+
+                this.updateLayout();
+
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    getFirstVisibleRenderedItem(): CardElement | undefined {
+        if (this.renderedElement && this._renderedPages?.length > 0) {
+            return this._renderedPages[0];
+        }
+        else {
+            return undefined;
+        }
+    }
+
+    getLastVisibleRenderedItem(): CardElement | undefined {
+        if (this.renderedElement && this._renderedPages?.length > 0) {
+            return this._renderedPages[this._renderedPages.length - 1];
+        }
+        else {
+            return undefined;
+        }
+    }
+
+    protected internalParse(source: any, context: SerializationContext) {
+        super.internalParse(source, context);
+
+        this._pages = [];
```

Comment:
nit: `timer` doesn't seem to be parsed.

Reply:
handled here:

https://github.com/microsoft/AdaptiveCards/pull/6667/files/6b34a781fd14f54c944e1df2b15136ce1ce07f4b#diff-c83b27dc841a20f8725b8b4aeb7f518336e1c5b77a9ae989824bb6908adc6204R64-R66